ui/components/ui/token-input/token-input.component.js

```diff
@@ -127,8 +130,8 @@ export default class TokenInput extends PureComponent {
       currency = currentCurrency;
       numberOfDecimals = 2;
     } else {
-      // Display ETH
-      currency = EtherDenomination.ETH;
+      // Display Native currency
+      currency = nativeCurrency;
```

Comment:
Here we notice this piece of code `if(showFiat){....} else{ fallback to ETH} ` . When you set Fiat in the settings=>General, the problem does not occur.
When you choose crypto, it falls back automatically to 'display ETH' and currency is `currency = EtherDenomination.ETH;`
regardless of the network you are on.



On the core repo; inside `TokenRatesController.ts` there is this fct `fetchAndMapExchangeRates `responsible for fetching the conversion rates. That function first checks if the `nativeCurrencySupported `by coingecko which in the case of BNB returns `true `and in the case of "MATIC" returns `false`.

If it is supported it will return the prices from coingecko, if it is not; it will 
1- fetch from coingecko the exchange rate of that token exp (USDC) against the fallback currency (eth)
2- fetch from cryptocompare the conversion of native currency exp (ETH) vs(MATIC)

And then eventually it will multiply them `tokenToVsCurrencyConversionRate * vsCurrencyToNativeCurrencyConversionRate` returning the conversion of USDC in the native currency (MATIC).